### ISSUE TITLE
[Feature Request] Impersonate start reply with

### ISSUE THREAD

**Issue by IkariDevGIT** [SillyTavern/SillyTavern#944]
**Have you searched for similar [requests](https://github.com/SillyTavern/SillyTavern/issues?q=)?**
Yes

**Is your feature request related to a problem? Please describe.**
When i press impersonate it often does not do what i want it to.

**Describe the solution you'd like**
I would like a checkbox somewhere, when it is enabled instead of replacing the content inside the textbox, it could continue it. if you dont like that you could simply turn it off.

**Describe alternatives you've considered**
I really dont have any easier soloutions, or alternatives.

**Additional context**
Not needed

**Comment by pabreu87** [comment]
I do this by typing the initial text of my intended action or dialogue and then pressing "Continue". AI will then continue my text.

**Comment by zkandell** [comment]
> I do this by typing the initial text of my intended action or dialogue and then pressing "Continue". AI will then continue my text.

Not the original person, but while that works, there are some issues with it. The way the interface works with it is a little odd (creating a message from you then continuing it on), and it will frequently not stop at the end of your message, but go on to impersonate the character as well, but within your message. The first could be mitigated by having the text fill out in the input box, the second would probably just take some stop sequences. 

That being said, I think that adding an option to the impersonate behavior to continue from rather than replace the text in the input box would be useful.

**Comment by pabreu87** [comment]
Yes it does impersonate the character within my message. I was hoping that would be a way to stop that.

**Comment by pabreu87** [comment]
> > I do this by typing the initial text of my intended action or dialogue and then pressing "Continue". AI will then continue my text.
> 
> , and it will frequently not stop at the end of your message, but go on to impersonate the character as well, but within your message. 

Yes it does impersonate the character within my message. I was hoping that there would be a way to stop that.

**Comment by Spacellary** [comment]
👀

This would be handy.

**Comment by Spacellary** [comment]
Yes.

**Comment by underscorex86** [comment]
this is effectively closed now that stscript exists, one can create buttons to /continue any {{input}} etc @Wolfsblvt

**Comment by Wolfsblvt** [comment]
That is the truth. That slash command supports reading from the input field since... when I started with ST.
We can close this.
